apps/web/src/screens/Dashboard/components/AptosTokensTab/PendingTokens.tsx

```diff
@@ -0,0 +1,107 @@
+import type { FC } from 'react';
+import { Image, StyleSheet } from 'react-native';
+import RequirePasscodeModal from '@walless/app/modals/RequirePasscodeModal';
+import { shortenAddress } from '@walless/core';
+import { type AptosPendingToken, aptosState } from '@walless/engine';
+import { BindDirections, Button, modalActions, Text, View } from '@walless/gui';
+import { RequestType } from '@walless/messaging';
+import { encryptedMessenger } from 'bridge/utils/messaging';
+import { useSnapshot } from 'valtio';
+
+interface Props {
+	fee: number;
+}
+
+const PendingTokens: FC<Props> = ({ fee }) => {
+	const tokens = Array.from(useSnapshot(aptosState).pendingTokens.values());
+
+	const timestampToDate = (timestamp: number) =>
+		new Date(timestamp).toLocaleString();
+
+	const handleClaimToken = async (token: AptosPendingToken) => {
+		modalActions.show({
+			component: () => (
+				<RequirePasscodeModal
+					title={`Claim ${token.name}`}
+					description={`Claim this token will submit an on-chain transaction and will require a ${fee} APT gas fee.`}
+					onPasscodeComplete={async (passcode) => {
+						const res = await encryptedMessenger.request('kernel', {
+							type: RequestType.CLAIM_TOKEN_ON_APTOS,
+							transaction: JSON.stringify({
+								pubkey: token.toAddress,
+								sender: token.fromAddress,
+								creator: token.creatorAddress,
+								collectionName: token.collectionName,
+								name: token.name,
+							}),
+							passcode,
+						});
+						return res;
+					}}
+					onActionComplete={() => {
+						modalActions.hide('passcode');
+					}}
+				/>
+			),
+			id: 'passcode',
+			bindingDirection: BindDirections.InnerBottom,
+		});
```

Comment:
could unify this action under `modalActions.showRequiredPasscodeModal`, because I see it appears multiple times